__tests__/allTests.test.js

```diff
@@ -0,0 +1,233 @@
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              article_img_url: expect.any(String),
+              comment_count: expect.any(Number)
+            })
+            //expect(body.topics).toHaveLength(3)
+      
+
+        });
+    })
+
+
+    test('GET:200 sends undefined to the client when making an invalid number request for the articles', () => {
+      return request(app)
+      .get('/api/articles/9999')
+      .expect(200)
+      .then(({body}) =>{
+          let obj = {body}
+
+          let obj2 = obj[0];
+
+        //  console.log(obj2)
+
+         expect(obj2).toBe(undefined)
+      });
+  })
+})
+
+
+
+
+describe('/api/articles/', () => {
+  test('GET:200 sends an object of topics to the client when making a valid request', () => {
+      return request(app)
+      .get('/api/articles/1')
+      .expect(200)
+      .then(({body}) =>{
+          let obj = body
+
+          let obj2 = obj[0]
+
+        //  console.log(obj2)
+
+         // console.log({body})
+
+          expect(obj2).toMatchObject({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String)
+          })
+          //expect(body.topics).toHaveLength(3)
+    
+
+      });
+  })
+})
+
+
+
+
+describe('/api/articles/:article_id/comments', () => {
+  test('GET:200 sends an object of topics to the client when making a valid request', () => {
+      return request(app)
+      .get('/api/articles/1/comments')
+      .expect(200)
+      .then(({body}) =>{
+          let obj = body
+
+          let obj2 = obj[0]
+
+          console.log(obj2)
+
+         // console.log({body})
+
+          expect(obj2).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: expect.any(Number),
+          })
+      });
+  })
+
+
+  test('sends status code 404 and undefined to the client when making an invalid article_id number request', () => {
+    return request(app)
+    .get('/api/articles/50000/comments')
+    .expect(404)
+    .then(({body}) =>{
+        let obj = body
+
+        let obj2 = obj[0]
+
+        console.log(obj2)
+
+       // console.log({body})
+
+        expect(obj2).toBe(undefined)
+    });
+})
+
+
+test('sends status code 404 and an empty object of topics to the client when making an invalid article_id request', () => {
+  return request(app)
+  .get('/api/articles/raj/comments')
+  .expect(400)
+  .then(({body}) =>{
+      let obj = body
+
+      let obj2 = obj[0]
+
+      console.log(obj2)
+
+     // console.log({body})
+
+      expect(obj2).toBe(undefined)
```

Comment:
You'd want to make an expect statement about the error message coming back, not that this property is undefined. We'd expect some kind of message telling the user that it was a bad request